SilverSolver/assignment1/Q2.py

```diff
@@ -0,0 +1,44 @@
+class single_linked_node:
+
+    def __init__(self, value=None, child=None):
+        self.child = child
+        self.value = value
+
+    def __getitem__(self, k):
+        k = -k
+        list_len = 0
+        current_node = self
+        while (current_node):
+            list_len += 1
+            current_node = current_node.child
+        element_num = list_len - k
+        if element_num < 0 or element_num > list_len:
+            raise IndexError('wrong k')
+        else:
+            current_num  = 0
+            current_node = self
+            while(current_node):
```

Comment:
Mmmh, interesting, Here, you start over, once more, from the beginning of the list.  SO what if the the list is very very long, you see where I am going?
Can we discuss a different implementation of that method that allows to only go through the list only once?

Reply:
I could keep in the memory k previous visited nodes of the singly linked list and return the k-th previous visited node when I reach the end of the list.
But it will consume more memory. So I don't know is it ok?